src/files-and-uploads/FileInfo.jsx

```diff
@@ -15,29 +17,60 @@ import {
   Truncate,
   IconButtonWithTooltip,
   CheckboxControl,
+  Spinner,
 } from '@edx/paragon';
 import { ContentCopy, InfoOutline } from '@edx/paragon/icons';
-import AssetThumbnail from './FileThumbnail';
 
+import { getFileSizeToClosestByte } from './data/utils';
+import { RequestStatus } from '../data/constants';
+import AssetThumbnail from './FileThumbnail';
 import messages from './messages';
 
 const FileInfo = ({
   asset,
   isOpen,
   onClose,
   handleLockedAsset,
+  usagePathStatus,
   // injected
   intl,
 }) => {
-  const [lockedState, setLockedState] = useState(asset.locked);
+  const [lockedState, setLockedState] = useState(asset?.locked);
   const handleLock = (e) => {
```

Comment:
I have a [unit test (FilesAndUploads.test.jsx Line 250)](https://github.com/openedx/frontend-app-course-authoring/blob/master/src/files-and-uploads/FilesAndUploads.test.jsx#L250) for this function. I have it in this file because it is exclusively used in this file and is just calling a series of functions already defined in the file.